mempool/clist_mempool.go

```diff
@@ -381,29 +406,46 @@ func (mem *CListMempool) addTx(memTx *mempoolTx, sender p2p.ID) bool {
 			memTx := elem.Value.(*mempoolTx)
 			if found := memTx.addSender(sender); found {
 				// It should not be possible to receive twice a tx from the same sender.
-				mem.logger.Error("tx already received from peer", "tx", tx.Hash(), "sender", sender)
+				mem.logger.Error("Tx already received from peer", "tx", tx.Hash(), "sender", sender)
 			}
 		}
 
 		mem.logger.Debug(
-			"transaction already in mempool, not adding it again",
+			"Transaction already in mempool, not adding it again",
 			"tx", tx.Hash(),
+			"lane", lane,
 			"height", mem.height.Load(),
 			"total", mem.Size(),
 		)
 		return false
 	}
 
+	// Get lane's clist.
+	txs, ok := mem.lanes[lane]
+	if !ok {
+		mem.logger.Error("Lane does not exist", "tx", log.NewLazySprintf("%v", tx.Hash()), "lane", lane)
+		return false
+	}
+
 	// Add new transaction.
 	_ = memTx.addSender(sender)
-	e := mem.txs.PushBack(memTx)
+	e := txs.PushBack(memTx)
+
+	// Update auxiliary variables.
 	mem.txsMap.Store(txKey, e)
+	mem.txLanes.Store(txKey, lane)
+
+	// Update size variables.
 	mem.txsBytes.Add(int64(len(tx)))
+	mem.numTxs.Add(1)
+
+	// Update metrics.
 	mem.metrics.TxSizeBytes.Observe(float64(len(tx)))
 
 	mem.logger.Debug(
-		"added valid transaction",
+		"Added transaction",
 		"tx", tx.Hash(),
+		"lane", lane,
 		"height", mem.height.Load(),
 		"total", mem.Size(),
```

Comment:
Do we want to also add the size of the lane to the Debug line?

Reply:
I'd say yes.